scripts/run.sh

```diff
@@ -23,9 +23,32 @@ then
     WEBVERSION=$4
 fi
 
+OS="nix"
 ENV_DIR="$OUTPUT_DIR/env"
 DOCKER_DIR="$OUTPUT_DIR/docker"
 
+# Initialize UID/GID which will be used to run services
+if ! grep -q "^LOCAL_UID=" $ENV_DIR/uid.env 2>/dev/null || ! grep -q "^LOCAL_GID=" $ENV_DIR/uid.env 2>/dev/null
+then
+    LUID="LOCAL_UID=`id -u $USER`"
+    LGID="LOCAL_GID=`id -g $USER`"
+    mkdir -p $ENV_DIR
+    echo $LUID >$ENV_DIR/uid.env
+    echo $LGID >>$ENV_DIR/uid.env
+fi
+
+# Up to Core version 1.19.0, UID/GID given by the user in uid.env may collide with existing UID/GID in images
+# We then must enforce a UID/GID pair known to be available in images
+# Newer versions properly use -o switch to useradd / groupadd to avoid this problem
+if [[ "$COREVERSION" == *.*.* ]] &&
+   echo -e "1.19.0\n$COREVERSION" | sort -t '.' -k 1,1 -k 2,2 -k 3,3 -n | awk 'END {if($0!="1.19.0") {exit 1}}'
```

Comment:
What if a user is coming from 1.18 or 1.17? Unfortunately some users are very outdated before they decide to update.

Reply:
This will work 👍 
`$COREVERSION` is the new version to use, compared to `1.19.0` (the last version where we should not enforce UID/GID).
`sort` sorts these 2, and `awk` returns false if `1.19.0` is not the greater version (the last line given by `sort` is the greater version, `awk` compares the last line to `1.19.0`).